packages/components/src/util/URLService.ts

```diff
@@ -13,19 +13,29 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { OrderedSet } from 'immutable';
+import { List, Map, OrderedSet } from 'immutable';
 
 import { AppURL } from '../url/AppURL';
 
 import { AppRouteResolver } from './AppURLResolver';
+import { ActionURL } from '@labkey/api';
+import { createProductUrl } from '../components/navigation/utils';
 
 const ADD_TABLE_ROUTE = 'application/routing/add-table-route';
 
 type RoutingTable = Map<string, string | boolean>;
 
 let resolvers = OrderedSet<AppRouteResolver>();
 
+export interface URLMapper {
+    resolve(url, row, column, schema, query): AppURL | string | boolean;
+}
+
 export namespace URLService {
+    export let urlMappers: List<URLMapper> = List<URLMapper>();
```

Comment:
It seems a little odd to be exporting a `let` variable. Do you think we should instead just add a getter for this? `getUrlMappers()`?